pyomo/solvers/plugins/solvers/GAMS.py

```diff
@@ -1039,6 +1054,49 @@ def solve(self, *args, **kwds):
         return results
 
 
+class OutputStream:
+    """Output stream object for simultaneously writing to multiple streams.
+
+    tee=False:
+        If set writing to this stream will write to stdout.
+    logfile=None:
+        Optionally a logfile can be written.
+
+    """
+
+    def __init__(self, tee=False, logfile=None):
+        """Initialize output stream object."""
+        if tee:
+            self.tee = sys.stdout
+        else:
+            self.tee = None
+        self.logfile = logfile
+        self.logfile_buffer = None
+
+    def __enter__(self):
+        """Enter context of output stream and open logfile if given."""
+        if self.logfile:
+            self.logfile_buffer = open(self.logfile, 'a')
+        return self
+
+    def __exit__(self, *args, **kwargs):
+        """Enter context of output stream and close logfile if necessary."""
+        if self.logfile_buffer:
+            self.logfile_buffer.close()
+        self.logfile_buffer = None
+
+    def write(self, message):
+        """Write messages to all streams."""
+        if self.tee:
+            self.tee.write(message)
+        if self.logfile_buffer:
+            self.logfile_buffer.write(message)
```

Comment:
I would rather these tests be against None and not rely on implicit casts to bool.  That is:
```
if self.tee is not None:
    self.tee.write(message)
if self.logfile_buffer is not None:
    self.logfile_buffer.write(message)
```
...the same applies to the other methods in this class.

Reply:
Done in dbcff25a